valohai_yaml/objs/pipelines/override.py

```diff
@@ -69,3 +72,38 @@ def serialize(self) -> OrderedDict:  # type: ignore[type-arg]
                     elide_empty_iterables=True,
                 )
         return val
+
+    @classmethod
+    def default_merge(cls, a: Override | None, b: Any) -> Override:
+        result = copy.deepcopy(a) if a else cls()
+        result.parameters = merge_dicts(
+            b.parameters,
+            result.parameters,
+            merger=merge_simple,
+            copier=copy.deepcopy,
+        )
+        result.inputs = merge_dicts(
+            b.inputs,
+            result.inputs,
+            merger=merge_simple,
+            copier=copy.deepcopy,
+        )
+        result.environment_variables = merge_dicts(
+            b.environment_variables,
+            result.environment_variables,
+            merger=merge_simple,
+            copier=copy.deepcopy,
+        )
+        return result
+
+    def normalize_to_template(self) -> OrderedDict:
+        override = self.serialize()
+        override["inputs"] = {
+            input["name"]: listify(input.get("default", None))
+            for input in override.get("inputs", [])
+        }
+        override["parameters"] = {
+            parameter["name"]: parameter.get("default", None)
+            for parameter in override.get("parameters", [])
+        }
```

Comment:
Again, this would probably be neater if it used the objects instead of just yoloing around with the serialized dicts :)

Something like

```suggestion
        inputs = {
            input.name: listify(input.default)
            for input in self.inputs
        }
        parameters = {
            parameter.name: parameter.default
            for parameter in override.parameters
            if parameter.default is not None
        }
```
or similar...?